www/script/project/rtTestReport.js

```diff
@@ -0,0 +1,193 @@
+/*global define, jQuery, require */
+define(function (require) {
+    "use strict";
+
+    var $ = require('jquery'),
+        helpers = require('helpers'),
+        moment = require('moment'),
+        oTable,
+        th,
+        rtTestReport,
+        realtimeRouting = require('realtimerouting');
+
+
+    rtTestReport = {
+        init: function () {
+            th = $('.table-holder');
+            rtTestReport.dataTablesInit();
+            rtTestReport.parseTimes("[data-time]");
+            rtTestReport.setupFilterButtons();
+            rtTestReport.addCodebasesBar();
+
+            var checkboxesList = $('#CheckBoxesList').find('input');
+            checkboxesList.bind("click.katana", function () {
+                rtTestReport.filterCheckboxes();
+            });
+            rtTestReport.filterCheckboxes();
+            helpers.initRecentBuildsFilters();
+
+            setTimeout(rtTestReport.addFailureButtons, 100);
+        },
+        dataTablesInit: function () {
+            //Filter tables based on checkboxes
+            $.fn.dataTableExt.oApi.fnFilterAll = function (oSettings, sInput, iColumn, bRegex, bSmart) {
+                var settings = $.fn.dataTableSettings,
+                    i;
+
+                for (i = 0; i < settings.length; i += 1) {
+                    settings[i].oInstance.fnFilter(sInput, iColumn, bRegex, bSmart);
+                }
+
+                $('.dataTables_empty').closest(th).hide();
+            };
+
+            //Filter when return is hit
+            jQuery.fn.dataTableExt.oApi.fnFilterOnReturn = function () {
+                var that = this;
+
+                this.each(function (i) {
+                    $.fn.dataTableExt.iApiIndex = i;
+                    var anControl = $('input', that.fnSettings().aanFeatures.f);
+                    anControl.unbind('keyup').bind('keypress.katana', function (e) {
+                        if (e.which === 13) {
```

Comment:
Maybe we should mention that keypress 13 is enter for those who don't remember it by heart. We could even have it as a global setting